fedimintd/src/ui/mod.rs

```diff
@@ -110,25 +109,9 @@ async fn deal(
 ) -> Result<Redirect, (StatusCode, String)> {
     let mut state = state.write().unwrap();
     state.federation_name = form.federation_name;
-    state.btc_rpc = Some(form.btc_rpc.clone());
-
-    let parts: Vec<&str> = form.btc_rpc.split('@').collect();
-    let user_pass = parts[0].to_string();
-    let btc_rpc_address = parts[1].to_string();
-    let parts: Vec<&str> = user_pass.split(':').collect();
-    let btc_rpc_user = parts[0].to_string();
-    let btc_rpc_pass = parts[1].to_string();
-    let btc_rpc = BitcoindRpcCfg {
-        btc_rpc_address,
-        btc_rpc_user,
-        btc_rpc_pass,
-    };
```

Comment:
😬

Reply:
Right, if we want to make it easily user-configurable the "enum in module (or server) cfg" approach might be better.

```rust
enum BitcoinBackend {
    Bitcoind { … /* connect data */ }
    Electrum { … /* connect data */ }
    Esplora { … /* connect data */ }
}
```

You could still create the bitcoin backend outside the module if you want to, but env vars are a hassle to deal with when everything else can be done via the UI.